tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py

```diff
@@ -199,6 +199,18 @@ def with_git(base_container: Container) -> Container:
 
     return base_container.with_exec(package_install_command + packages_to_install)
 
+def with_packages(base_container: Container, packages_to_install) -> Container:
+    """Installs  in a alpine based container.
+    Args:
+        context (Container): A alpine based container.
```

Comment:
```suggestion
        base_container (Container): A alpine based container.
```
I'm not sure its an alpine based container. Isn't the package manager `apk` on Alpine?
I like this new function but I'm a bit worried it won't work for any container you pass in depending of its base image...

Reply:
Ah sorry I saw that "python:3-alpine" was used when calling

But I did not see the `slim` default.

We should probably pick one type of linux distro and stick to it.

I would vote for debian for the same reasons in [this post](https://medium.com/swlh/alpine-slim-stretch-buster-jessie-bullseye-bookworm-what-are-the-differences-in-docker-62171ed4531d)

> However, some teams are moving away from alpine because these images can cause compatibility issues that are hard to debug. Specifically, if using python images, some wheels are built to be compatible with Debian and will need to be recompiled to work with an Apline-based image.

Ive already ran into this in another PR

Reply:
> What do you think about improving the with_python_base environment and pass optional packages_to_install?

I think in general im against? I really like the composability of what you have set up here. as its very chainable

I know right now its a bit annoying defining all the variables
```py
python_container = with_python_base(context)
python_with_packages = with_packages(python_container, ["git"])
python_with_packages_poetry = with_poetry(python_with_packages)
# etc...
```

But I think its the right direction because with a little bit of work we could extend this pattern to work with a chain function
```py
container = chain_functions(
    context,
    with_python_base,
    with_packages(["git"]),
    with_poetry,
)
```

or start using these like class methods

```py
context.with_python_base().with_packages(["git"]).with_poetry()
```

I think these are better than ending up with a lot of arguments to `with_python_base`

```py
with_python_base(
    context,
    includes=[],
    excludes=[],
    system_packages=[],
    pip_packages=[],
    local_poetry_packages=[],
)
```

But thats an opinion! 😅